src/neuroglancer/ui/context_menu.ts

```diff
@@ -30,7 +30,7 @@ export function positionContextMenu(menu: HTMLElement, event: MouseEvent) {
       document.documentElement!.scrollTop + Math.min(viewportHeight - offsetHeight, event.clientY);
   menu.style.left = posX + 'px';
   menu.style.top = posY + 'px';
-  menu.style.visibility = null;
```

Comment:
What problem is this change addressing?  The reason for using visibility hidden rather than display none is that  this code checks offsetWidth and offsetHeight of the menu, and I believe those are 0 if display is none.

Reply:
Hidden does not remove it from the page. And when you embed it in a larger application that breaks the flow of the layout with a large white space, when the menu is not shown.

Reply:
positioning and width seemed to be fine, with this change, in my testing on Chrome.

Reply:
Okay.  It should be set to display: none when not supposed to be shown, but we do need to set it to visibility hidden, display normal, just before positionContextMenu, so that offsetWidth and offsetHeight are valid.  As you have it currently, if you try to open the context menu very close to the right edge of the screen, it doesn't display properly.

Reply:
I just moved the "menu.style.display = 'block';" to before the offset check and that works in firefox and chrome.